pwndbg/aglib/regs.py

```diff
@@ -125,7 +125,8 @@ def __setattr__(self, attr: str, val: Any) -> None:
         if attr in ("last", "previous"):
             super().__setattr__(attr, val)
         else:
-            pwndbg.dbg.selected_frame().reg_write(attr, int(val))
+            if not pwndbg.dbg.selected_frame().reg_write(attr, int(val)):
```

Comment:
Or inside here we should "remap" if attr == `pc` to correct real register?

Reply:
And what about `sp` register? Like we have in `fix()` method:
```py
    def fix(self, expression: str) -> str:
        for regname in set(self.all + ["sp", "pc"]):
            expression = re.sub(rf"\$?\b{regname}\b", r"$" + regname, expression)
        return expression
```

Reply:
> Maybe reg_write should only accept real registers?

Probably not. I think at least `pc` and `sp` are used very widely in Pwndbg.

> Or inside here we should "remap" if attr == pc to correct real register?

Personally, I think mapping `pc` to `regs_sets[...].pc` and still doing the sanity check against the registers listed in the reg sets is a cleaner solution, because that means we only ever have to deal with real register names in the Debugger API. Buuuut, both GDB and LLDB accept `pc` as a register name just fine, so it's not like this won't work.

> And what about sp register?

Yeap, this will probably need an exception for `sp`, too..